mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala

```diff
@@ -197,8 +229,92 @@ class LocalLDAModel private[clustering] (
   // TODO:
   // override def topicDistributions(documents: RDD[(Long, Vector)]): RDD[(Long, Vector)] = ???
 
+  /**
+   * Calculate and return log variational bound on perplexity using the provided `documents` of
+   * documents as a test corpus.
+   * Perplexity on a test corpus is calculated as:
+   *    perplexity(documents) = exp( -log p(documents) / numWords )
+   * where p is the LDA model. It is upper bounded by the variational distribution using:
+   *    perplexity(documents) <= exp( -(E_q[log p(documents] - E_q[log q(documents)]) / numWords }
```

Comment:
Now that I see it, this description seems too simplified to be useful.  E.g., "p(documents)" and "q(documents)" are misleading since they are actually distributions over different sets of random variables.  But including full details sounds too complex.  How about keeping lines 233-235 and replacing the equations with a reference to Eq 16 in the Online LDA paper?

Also, state that this is returning the log perplexity per word.